src/status.hpp

```diff
@@ -50,19 +50,19 @@ enum class StatusCode {
     MODEL_VERSION_POLICY_UNSUPPORTED_KEY, /*!< Model version policy contains invalid key */
     GRPC_CHANNEL_ARG_WRONG_FORMAT,
-    CONFIG_FILE_TIMESTAMP_READING_FAILED,    /*!< Reading config file timestamp failed */
-    NO_MODEL_VERSION_AVAILABLE,              /*!< No model version found in path */
-    RESHAPE_ERROR,                           /*!< Impossible to perform reshape */
-    RESHAPE_REQUIRED,                        /*!< Model instance needs to be reloaded with new shape */
-    BATCHSIZE_CHANGE_REQUIRED,               /*!< Model instance needs to be reloaded with new batch size */
-    FORBIDDEN_MODEL_DYNAMIC_PARAMETER,       /*!< Value of the provided param is forbidden */
-    ANONYMOUS_FIXED_SHAPE_NOT_ALLOWED,       /*!< Anonymous fixed shape is invalid for models with multiple inputs */
-    ANONYMOUS_FIXED_LAYOUT_NOT_ALLOWED,      /*!< Anonymous fixed layout is invalid for models with multiple inputs */
-    CONFIG_SHAPE_IS_NOT_IN_NETWORK,          /*!< Configured tensor shape is not present in network */
-    CONFIG_LAYOUT_IS_NOT_IN_NETWORK,         /*!< Configured tensor layout is not present in network */
-    CONFIG_SHAPE_MAPPED_BUT_USED_REAL_NAME,  /*!< Using old name of input/output in config shape when mapped in mapping_config.json*/
-    CONFIG_LAYOUT_MAPPED_BUT_USED_REAL_NAME, /*!< Using old name of input/output in config layout when mapped in mapping_config.json*/
-    CANNOT_LOAD_NETWORK_INTO_TARGET_DEVICE,  /*!< Cannot load network into target device */
+    CONFIG_FILE_TIMESTAMP_READING_FAILED,          /*!< Reading config file timestamp failed */
+    NO_MODEL_VERSION_AVAILABLE,                    /*!< No model version found in path */
+    RESHAPE_ERROR,                                 /*!< Impossible to perform reshape */
+    RESHAPE_REQUIRED,                              /*!< Model instance needs to be reloaded with new shape */
+    BATCHSIZE_CHANGE_REQUIRED,                     /*!< Model instance needs to be reloaded with new batch size */
+    FORBIDDEN_MODEL_DYNAMIC_PARAMETER,             /*!< Value of the provided param is forbidden */
+    ANONYMOUS_FIXED_SHAPE_NOT_ALLOWED,             /*!< Anonymous fixed shape is invalid for models with multiple inputs */
+    ANONYMOUS_FIXED_LAYOUT_NOT_ALLOWED,            /*!< Anonymous fixed layout is invalid for models with multiple inputs */
+    CONFIG_SHAPE_IS_NOT_IN_NETWORK,                /*!< Configured tensor shape is not present in network */
+    CONFIG_LAYOUT_IS_NOT_IN_NETWORK,               /*!< Configured tensor layout is not present in network */
+    CONFIG_SHAPE_MAPPED_BUT_USED_REAL_NAME,        /*!< Using old name of input/output in config shape when mapped in mapping_config.json*/
+    CONFIG_LAYOUT_MAPPED_BUT_USED_REAL_NAME,       /*!< Using old name of input/output in config layout when mapped in mapping_config.json*/
+    CANNOT_LOAD_COMPILED_MODEL_INTO_TARGET_DEVICE,
```

Comment:
CANNOT_LOAD_MODEL_INTO_TARGET_DEVICE would fit better here. Model becomes CompiledModel only if the loading into device succeeds.

Reply:
We do not load CompiledModel into target device. We load Model into target device.

Reply:
we load ov::CompiledModel into target device:
https://github.com/openvinotoolkit/model_server/blob/atobisze_cleanup/src/modelinstance.cpp#L614

Reply:
We get CompiledModel as a result of loading Model into target device:
https://github.com/openvinotoolkit/model_server/blob/atobisze_cleanup/src/modelinstance.cpp#L592

Reply:
We first create model from files for OV:
        ``model = loadOVModelPtr(modelFile);``
        https://github.com/openvinotoolkit/model_server/blob/69e2b070c456099fc4486978962faa03adf7d0d5/src/modelinstance.cpp#L531
Here there is no target device.

Then we get compiled model by compiling it onto target device in:
https://github.com/openvinotoolkit/model_server/blob/atobisze_cleanup/src/modelinstance.cpp#L592

See also where this error is returned:
https://github.com/openvinotoolkit/model_server/blob/69e2b070c456099fc4486978962faa03adf7d0d5/src/modelinstance.cpp#L616

It is returned when we failed to load compile model.

Maybe we should alter the naming of those functions as well to:

Read model -> load model files // which does only load the layers and weights into RAM but not on inference device
Compile(?Load) model -> compile model onto target device // which compiles model & loads into inference device

Reply:
Everything you have stated above is true, but summarizes to the current wrong status code name:
```
CANNOT_LOAD_COMPILED_MODEL_INTO_TARGET_DEVICE
```
Cannot load compiled model into target device - means we HAVE compiled model and we pass it to device to load it. This is not true, we have MODEL and pass it to device and take COMPILED MODEL in return.

You can also see that in API 1.0 we had:
```
CANNOT_LOAD_NETWORK_INTO_TARGET_DEVICE
```
but not `CANNOT_LOAD_EXECUTABLE_NETWORK_INTO_TARGET_DEVICE` which would follow your naming convention used right now in this MR.

This is really minor, but I suggest changing it to:
```
CANNOT_LOAD_MODEL_INTO_TARGET_DEVICE
```
and the status message as well to match what it exactly does under the hood